src/prommis/precipitate/precipitator.py

```diff
@@ -255,6 +257,16 @@ def precipitate_generation(blk, t, comp):
                 / prop_s.stoich[comp]
             )
 
+        @self.Constraint(prop_aq.dissolved_elements)
```

Comment:
@andrewlee94 do you think this calculation should be done in the unit model or the properties?

Reply:
That depends on exactly what `E_D` and `N_D` are. If this is an equilibrium correlation (which I suspect it is) it is probably best done in the properties (or at least more standard and modular).

Looking at the equation as well, my first reaction was to ask if there was a better formulation, but I think you might have already thought about that (assuming `N_D` is a fixed variable/parameter).